DualMode-GRUBsetup.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+set -euo pipefail
+
+# Re-Install RPMs as necessary
+dnf -y reinstall grub2-pc
+
+# Move /boot/efi/EFI/redhat/grub.cfg as necessary
+[[ -e /boot/efi/EFI/redhat/grub.cfg ]] && \
+  mv /boot/efi/EFI/redhat/grub.cfg /boot/grub2
+
+# Make our /boot-hosted GRUB2 grub.cfg file
+grub2-mkconfig -o /boot/grub2/grub.cfg
+
+# Nuke grubenv file as necessary
+[[ -e /boot/grub2/grubenv ]] && rm -f /boot/grub2/grubenv
+
+# Create fresh grubenv file
+grub2-editenv /boot/grub2/grubenv create
+
+# Populate fresch grubenv file
+while read -r line
+do
+  key="$( echo "$line" | cut -f1 -d'=' )"
+  value="$( echo "$line" | cut -f2- -d'=' )"
+  grub2-editenv /boot/grub2/grubenv set "${key}"="${value}"
+  done <<< "$( grub2-editenv /boot/efi/EFI/redhat/grubenv list )"
```

Comment:
Might be nice to leave a comment on what this loop is doing, and why it's necessary. Feels pretty obsure.

Reply:
The line:
```bash
while read -r line
```
Combined with the line:
```bash
done <<< "$( grub2-editenv /boot/efi/EFI/redhat/grubenv list )"
```
Seemed self-explanatory?

Reply:
Lol, it was more the _why_ that wasn't immediately apparent and required some interpretation or prior experience. The new comment helps. The source is for EFI, the target is for BIOS. Got it.